### ISSUE TITLE
(failed) An LRS will reject a Cross Origin Request which attempts to send attachment data with error code 400 Bad Request (Communication 1.3.s3.b15) 

### ISSUE THREAD

**Issue by garemoko** [adlnet-archive/grail-issue-tracker#15]
I'm not sure what "attempts to send attachment data" entails here. It is literally not possible to send attachment data in this way. 

See https://www.youtube.com/watch?v=BKorP55Aqvg

**Comment by rchadwic** [comment]
Not sure I understand? What prevents me from sending a POST with CORS headers and a payload?

**Comment by garemoko** [comment]
I was thinking about binary attachments. You can send other types. Closing this.

**Comment by garemoko** [comment]
Re-openning. So given that it is possible to send non-binary attachments why is 400 expected?

**Comment by rchadwic** [comment]
Yeah, not sure I understand this. 1.3 is the alternate syntax for older browsers - maybe this is intended to be "in the case where the verb in the querystring is not "post" "

**Comment by bscSCORM** [comment]
I was about to open this as a separate issue against  "https://github.com/adlnet/lrs-conformance-test-suite", but I think this open issue covers my concern well enough.

It's important to note that where "Attachment Data" is used in the spec, it specifically means the additional multipart parts that are sent which contain the attachment itself, not the `attachments` property in the statement.

So I'm surprised to see the following test failure rationale, because *note* -- the test is not sending any attachment *data* at all, only the attachments property in the statement:
```
FAILED: An LRS will reject a Cross Origin Request which attempts to send attachment data with error code 400 Bad Request (Communication 1.3.s3.b14)
[ 'Communication.md#1.3.s3.b14' ]
RejectionError: expected 400 "Bad Request", got 204 "No Content"
REQUEST SUPERTEST
_______________________________________
POST /ScormEngineInterface/TCAPI/default/statements?method=PUT HTTP/1.1
Host: localhost:64619
Accept-Encoding: gzip, deflate
User-Agent: node-superagent/1.8.5
Content-Type: application/x-www-form-urlencoded
X-Experience-API-Version: 1.0.3
Authorization: Basic c2Nvcm06c2Nvcm0yMDA0
Content-Length: 459
Connection: close

"statementId=04378873-6149-4301-9fad-a0b33092434f&content={\"actor\":{\"objectType\":\"Agent\",\"name\":\"xAPI account\",\"mbox\":\"mailto:xapi@adlnet.gov\"},\"verb\":{\"id\":\"http://adlnet.gov/expapi/verbs/attended\",\"display\":{\"en-GB\":\"attended\",\"en-US\":\"attended\"}},\"object\":{\"objectType\":\"Activity\",\"id\":\"http://www.example.com/meetings/occurances/34534\"},\"attachments\":[]}&Content-Type=application/json&X-Experience-API-Version=1.0.2&Authorization=Basic c2Nvcm06c2Nvcm0yMDA0"
```

I also wonder if this could be another facet of the test suite's behavior of considering empty objects / collections invalid, which I see no justification for.  see: https://github.com/adlnet/lrs-conformance-test-suite/issues/129

**Comment by cr8onski** [comment]
fixed in [pr #126](https://github.com/adlnet/lrs-conformance-test-suite/pull/126)
